src/rewards/poprewards_cache.cpp

```diff
@@ -0,0 +1,126 @@
+// Copyright (c) 2019-2020 Xenios SEZC
+// https://www.veriblock.org
+// Distributed under the MIT software license, see the accompanying
+// file LICENSE or http://www.opensource.org/licenses/mit-license.php.
+
+#include <veriblock/rewards/poprewards_cache.hpp>
+
+namespace altintegration {
+
+PopRewardsBigDecimal PopRewardsCache::calculateDifficulty(
+    const index_t& /* ignore */) const {
+  auto it = buffer.crbegin();
+  // remove last element - it is the current block that we calculate difficulty
+  // for
+  ++it;
+  PopRewardsBigDecimal sumscore = 0.0;
+  size_t count = 0;
+  uint64_t difficultyInterval =
+      altParams_->getRewardParams().difficultyAveragingInterval();
+  for (; it != buffer.crend(); ++it) {
```

Comment:
need a check that we are summing no less than difficultyAveragingInterval scores unless we're close to the genesis/bootstrap block